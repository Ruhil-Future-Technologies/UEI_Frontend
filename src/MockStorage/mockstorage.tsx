/* eslint-disable @typescript-eslint/no-explicit-any */
import { Box } from '../Pages/AcademicHistory/AcademicHistory';

export const mockStorage: any = {
  menulist1: JSON.stringify([
    {
      created_at: 'Tue, 11 Jun 2024 12:15:27 GMT',
      form_data: {
        form_url: '/main/SuperAdmin',
        is_save: true,
        is_search: true,
        is_update: true,
      },
      id: 23,
      is_active: 1,
      is_deleted: false,
      menu_name: 'Menu',
      priority: 1,
      submenus: [
        {
          created_at: 'Mon, 24 Jun 2024 09:58:08 GMT',
          id: 56,
          is_active: 1,
          is_deleted: false,
          menu_master_id: 23,
          menu_master_name: 'User Authorization',
          menu_name: 'Menu',
          priority: 2,
          updated_at: 'Mon, 24 Jun 2024 13:37:35 GMT',
        },
      ],
      updated_at: 'Fri, 14 Jun 2024 07:40:04 GMT',
    },
    {
      created_at: 'Tue, 11 Jun 2024 12:15:27 GMT',
      form_data: {
        form_url: '/main/SuperAdmin',
        is_save: true,
        is_search: true,
        is_update: true,
      },
      id: 24,
      is_active: 1,
      is_deleted: false,
      menu_name: 'Submenu',
      priority: 1,
      submenus: [
        {
          created_at: 'Mon, 24 Jun 2024 09:58:08 GMT',
          id: 57,
          is_active: 1,
          is_deleted: false,
          menu_master_id: 24,
          menu_master_name: 'User Authorization',
          menu_name: 'Submenu',
          priority: 2,
          updated_at: 'Mon, 24 Jun 2024 13:37:35 GMT',
        },
      ],
      updated_at: 'Fri, 14 Jun 2024 07:40:04 GMT',
    },
  ]),
};

export const contextValue = {
  namepro: { id: 73, user_type: 'admin', userid: 'ashish7080@gmail.com' },
  setNamepro: jest.fn(),
  logoutpro: jest.fn(),
  setProImage: jest.fn(),
  proImage: '',
  setProPercentage: jest.fn(),
  ProPercentage: '',
  setNamecolor: jest.fn(),
  namecolor: '#ffffff',
  activeForm: 0, // Assign a number value, e.g., 0
  setActiveForm: jest.fn(), // Mock function remains unchanged
};

export const mockBoxes: Box[] = [
  {
    id: 334,
    institute_type: 'college',
    board: '',
    state_for_stateboard: '',
    institute_id: '95',
    course_id: 87,
    learning_style: 'online',
    class_id: '24',
    year: '2015-12-31T18:30:00.000Z',
    stream: '',
    university_id: '',
    sem_id: 1,
  },
];

export const mockClassValue = {
  data: [
    {
      class_name: 'class_12',
      created_at: 'Thu, 12 Dec 2024 11:43:04 GMT',
      created_by: 'afasfsafsaf',
      id: '5500819c-09e5-4543-8cf2-8f0d5cf6ffc3',
      is_active: true,
      is_deleted: false,
      updated_at: 'Thu, 12 Dec 2024 11:43:04 GMT',
      updated_by: 'Admin',
    },
  ],
  status: 200,
  message: 'Class Found Successfully',
};

export const mockbasicinfoPayload = {
  payload: [
    {
      student_login_id: '123',
      first_name: 'John',
      last_name: 'Patel',
      gender: 'male',
      dob: '01/01/2024',
      father_name: 'father',
      mother_name: 'mother',
      guardian_name: 'guardian',
      pic_path: 'pic_path',
      aim: 'aim',
    },
  ],
};
