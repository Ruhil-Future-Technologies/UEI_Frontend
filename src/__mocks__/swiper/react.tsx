import React from 'react';

export const Swiper: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => <div>{children}</div>;

export const SwiperSlide: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => <div>{children}</div>;
