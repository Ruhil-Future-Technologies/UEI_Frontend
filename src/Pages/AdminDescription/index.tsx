import * as React from 'react';

import { Card, CardContent,  SelectChangeEvent,  TextField } from '@mui/material';

import { useState, useEffect, useRef } from 'react';
import useApi from '../../hooks/useAPI';
import { toast } from 'react-toastify';
import { useNavigate } from 'react-router-dom';
import { Field, Form, Formik, FormikProps, setNestedObjectValues } from 'formik';
import * as Yup from 'yup';
import { deepEqual, inputfieldtext } from '../../utils/helpers';
import NameContext from '../Context/NameContext';
interface IAdminDescription {
    description: string
 }


// console.log(adminId);

export default function AdminDescription() {
    const initialState = {
        description: "",
       
    };
    const context = React.useContext(NameContext);
    const {namecolor }:any = context;
    let adminId = localStorage.getItem('_id')
    const { getData, postData, putData,loading } = useApi();
    const [description, setDesctiption] = useState(initialState);
    const [editFalg,setEditFlag]=useState<boolean>(false);
    const navigator = useNavigate()
    // const formRef = useRef() as any
    const formRef = useRef<FormikProps<IAdminDescription>>(null)

    const getDescription=async ()=>{
        try {
            const response=await  getData('admin_profile_description/edit/' + adminId);

            if (response && response?.status === 200) {
                setDesctiption(response?.data);
            }else if(response && response?.status===404){
               setEditFlag(true);
            } else {
                console.error("Unexpected response:", response);
            }
        } catch (error:any) {
            if(error?.response && error?.response?.status===401){
                navigator("/")
                toast.warning(("Please login again"),{
                    hideProgressBar:true,
                    theme:"colored"
                })
            }else{
                toast.error(("Request failed"),{
                    hideProgressBar:true,
                    theme:"colored"
                })
            }
        }
    }
    useEffect(() => {
        getDescription();
       
    }, [adminId])
    // const handleChange = async (e: any) => {
    //     const { name, value } = e.target;
    //     formRef?.current?.setFieldValue(e.target.name, e.target.value);
    //     const err = await formRef?.current?.validateForm()
    //     if (err && Object.keys(err).length > 0) {
    //         formRef?.current?.setErrors(err)
    //         formRef?.current?.setTouched(setNestedObjectValues(err, true))
    //         formRef?.current?.setFieldError(name, formRef?.current?.errors?.[name as keyof IAdminDescription])
    //         formRef?.current?.setFieldTouched(name, true)
    //     }
    //     setDesctiption(value);
    // };
    const handleChange = async (e: React.ChangeEvent<HTMLInputElement> | SelectChangeEvent<string>, fieldName: string) => {
        setDesctiption((prevMenu) => {
            return {
                ...prevMenu,
                [e.target.name]: e.target.value,
            };
        });
        formRef?.current?.setFieldValue(fieldName, e.target.value);
        await formRef?.current?.validateField(fieldName)
        if (formRef?.current?.errors?.[fieldName as keyof IAdminDescription] !== undefined) {
            formRef?.current?.setFieldError(fieldName, formRef?.current?.errors?.[fieldName as keyof IAdminDescription])
            formRef?.current?.setFieldTouched(fieldName, true)
        }
    };
    // const submitHandle = (event: any,descriptionData:{description:string}) => {
        const submitHandle = async (description1: IAdminDescription) => {
            const eq = deepEqual(description1,formRef?.current?.initialValues)
        // event.preventDefault();
        let paylod = {
            admin_id: adminId,
            description: description1?.description
        }
        if(editFalg){
            const saveData=async ()=>{
                try {
                    const response=await postData('admin_profile_description/add', paylod);
          
                    if (response?.status === 200) {
                        toast.success("Admin description saved successfully", {
                            hideProgressBar: true,
                            theme: "colored"
                        })
                    }
                } catch (error:any) {
                    if(error?.response?.status === 401){
                        toast.warning(("Please login again"),{
                            hideProgressBar:true,
                            theme:"colored"
                        })
                    }else{
                        toast.error(("Request failed"),{
                            hideProgressBar:true,
                            theme:"colored"
                        })
                    }
                }
            }
         saveData();
           
        }else if(!editFalg){
            const editData=async ()=>{
                try {
                    const response=await putData('admin_profile_description/edit/'+adminId,paylod);
        
                    if(response?.status === 200){
                         toast.success("Admin description updated successfully",{
                            hideProgressBar:true,
                            theme:"colored"
                         })
                         getDescription()
                    }else{
                        toast.error(("something want wrong "),{
                            hideProgressBar:true,
                            theme:"colored"
                        })
                    } 
                } catch (error:any) {
                    if(error?.response?.status===401){
                        navigator("/")
                        toast.warning(("Please login again"),{
                            hideProgressBar:true,
                            theme:"colored"
                        })
                    }else{
                        toast.error(("Request failed"),{
                            hideProgressBar:true,
                            theme:"colored"
                        })
                    }
                }
             
            }

                 // eslint-disable-next-line no-lone-blocks
                 {!eq && (

                     editData()
                 )}
         
        }
       
    }
     const descriptionSchema = Yup.object().shape({
        description: Yup.string()
            .required("Please enter Description")
    })
    return (

          
            <Formik
        onSubmit={(formData) => submitHandle(formData)}
        initialValues={{
            description:description?.description
        }}
        enableReinitialize
        validationSchema={descriptionSchema}
        innerRef={formRef}
    >
        {({ errors, values ,touched , isValid, dirty}:any) => (
            <Form>

                <Card className='description' style={{ margin: "15px" }}>

                    <CardContent className='description'>
                        {/* <TextField
                            id="description"
                            label="Description *"
                            multiline
                            rows={9}
                            fullWidth
                            margin="normal"
                            value={values.description}
                            onChange={handleChange}
                        /> */}
                        
                            <Field
                                id="description"
                                label="Description *"
                                component={TextField}
                                type="text"
                                name="description"
                                fullWidth
                                rows={9}
                                multiline
                                margin="normal"
                                value={values?.description}
                                onChange={(e: React.ChangeEvent<HTMLInputElement>) => handleChange(e, "description")}
                                InputProps={{
                                    style: {
                                        color: inputfieldtext(namecolor), // Change this to your desired text color
                                    },
                                }}
                                InputLabelProps={{
                                    style: {
                                        color:inputfieldtext(namecolor), // Change this to your desired label color
                                    },
                                }}
                                
                            />
                            {touched?.description && errors?.description ?
                                <p style={{ color: 'red' }}>{errors?.description}</p> : <></>
                                            }
                    </CardContent>
                </Card>
                <div className='row justify-content-center' style={{margin:"10px"}}>

                    <div className="col-2">
                        <button className='btn btn-primary  mainbutton'  >{editFalg?"save":"save changes"}</button>
                    </div>

                </div>
            </Form>
        )}
        </Formik>
        
 
    );
}
