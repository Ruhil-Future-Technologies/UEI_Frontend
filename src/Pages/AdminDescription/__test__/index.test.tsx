import { fireEvent, render } from "@testing-library/react";
import React from "react";
import NameContext from "../../Context/NameContext";
import { contextValue } from "../../../MockStorage/mockstorage";
import { MemoryRouter } from "react-router-dom";
import AdminDescription from "..";
import useApi from "../../../hooks/useAPI";
jest.mock('../../../hooks/useAPI', () => ({
    __esModule: true,
    default: jest.fn(() => ({
      getData: jest.fn(),
      postData: jest.fn(),
      putData: jest.fn(),
      postFileData:jest.fn()
    })),
  }));
  
  jest.mock('react-toastify', () => ({
    toast: {
      success: jest.fn(),
      error: jest.fn(),
    },
  }));

  const mockGetData = jest.fn();
    const mockPostData = jest.fn();
    const mockPutData = jest.fn();
    beforeEach(() => {
      mockPostData.mockReset();
      mockPutData.mockReset();
      (useApi as jest.Mock).mockImplementation(() => ({
        getData: mockGetData.mockResolvedValue({ getData: mockGetData }), // Mock resolved promise
        postData: mockPostData,
        putData: mockPutData, 
      }));
    });
describe("Admin Description page component", () => {

    const mockSetActiveForm = jest.fn();
    const mockActiveForm = 3;
    const renderedComponent=()=>{
        return render(<NameContext.Provider value={contextValue}>
            <MemoryRouter>
                <AdminDescription
                activeForm={mockActiveForm}
                setActiveForm={mockSetActiveForm}
                />
            </MemoryRouter>
        </NameContext.Provider>)
    }

    it("should render the page properly", () => {
        const {asFragment}=renderedComponent();
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render the fileds correctly", () => {
        const {getByTestId}=renderedComponent();
        const description=getByTestId("description");
        expect(description).toBeInTheDocument();
    });

    it("should handlecahnges function working at onchange ",()=>{
        const {getByLabelText}=renderedComponent();
        const descriptionInput=getByLabelText("Description *") as HTMLInputElement;
        fireEvent.change(descriptionInput, {target: {value: "New Description"}});
        expect(descriptionInput.value).toBe("New Description");
    })

   
})