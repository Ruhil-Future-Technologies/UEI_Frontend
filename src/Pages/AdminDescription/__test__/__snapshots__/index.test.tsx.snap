// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Description page component should render the page properly 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      data-testid="description"
      field="[object Object]"
      form="[object Object]"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="description"
        id="description-label"
      >
        Description *
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          id="description"
          rows="9"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Description *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="row justify-content-center"
      style="margin: 10px;"
    >
      <div
        class="col-lg-12"
      >
        <div
          class="mt-3 d-flex align-items-center justify-content-between"
        >
          <button
            class="btn btn-outline-dark prev-btn px-lg-4  rounded-pill"
            type="button"
          >
            Previous
          </button>
          <button
            class="btn btn-dark px-lg-5 px-4  ms-auto d-block rounded-pill next-btn"
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
